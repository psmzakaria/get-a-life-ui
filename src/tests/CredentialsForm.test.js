import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import CredentialsForm from "./../components/CredentialsForm";

it("should render the sign in button", () => {
  const state = {
    componentToDisplay: "Sign Up",
    signUpError: {
      hidden: true,
      header: "",
      content: ""
    },
    signInError: {
      hidden: true,
      header: "",
      content: ""
    },
    username: "",
    password: ""
  };

  const renderer = new ShallowRenderer();
  renderer.render(<CredentialsForm state={state} />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
