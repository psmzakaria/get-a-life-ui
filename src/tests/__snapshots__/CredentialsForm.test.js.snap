// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the sign in button 1`] = `
<div
  className="ui inverted segment form"
>
  <h1>
    Sign Up
  </h1>
  <Form
    as="form"
  >
    <Message
      content=""
      header=""
      hidden={true}
      negative={true}
    />
    <FormField>
      <label>
        Username
      </label>
      <input
        id="username"
        onChange={undefined}
        placeholder="Username"
        value=""
      />
    </FormField>
    <FormField>
      <label>
        Password
      </label>
      <input
        id="password"
        onChange={undefined}
        placeholder="Password"
        type="password"
        value=""
      />
    </FormField>
    <SignUpButton
      handleOnSubmit={undefined}
      loadComponent={undefined}
    />
  </Form>
</div>
`;
